node/sc/bridge_addr_journal.go

```diff
@@ -24,31 +24,38 @@ import (
 	"errors"
 	"io"
 	"os"
+	"strings"
 
 	"github.com/klaytn/klaytn/common"
 	"github.com/klaytn/klaytn/node/sc/bridgepool"
 	"github.com/klaytn/klaytn/rlp"
 )
 
 var (
-	errNoActiveAddressJournal = errors.New("no active address journal")
-	errDuplicatedJournal      = errors.New("duplicated journal is inserted")
-	errEmptyBridgeAddress     = errors.New("empty bridge address is not allowed")
+	ErrNoActiveAddressJournal = errors.New("no active address journal")
+	ErrDuplicatedJournal      = errors.New("duplicated journal is inserted")
+	ErrDuplicatedAlias        = errors.New("duplicated alias")
+	ErrEmptyBridgeAddress     = errors.New("empty bridge address is not allowed")
+	ErrEmptyJournalCache      = errors.New("empty bridge journal")
+	ErrEmptyBridgeAlias       = errors.New("empty bridge Alias")
+	ErrNotAllowedAliasFormat  = errors.New("Not allowed bridge alias format")
 )
 
 // bridgeAddrJournal is a rotating log of addresses with the aim of storing locally
 // created addresses to allow deployed bridge contracts to survive node restarts.
 type bridgeAddrJournal struct {
-	path   string         // Filesystem path to store the addresses at
-	writer io.WriteCloser // Output stream to write new addresses into
-	cache  map[common.Address]*BridgeJournal
+	path       string         // Filesystem path to store the addresses at
+	writer     io.WriteCloser // Output stream to write new addresses into
+	cache      map[common.Address]*BridgeJournal
+	aliasCache map[string]common.Address
```

Comment:
It seems that the concurrent access error happens if more than two APIs are concurrently requested. What do you think about adding mutex for these maps?

Reply:
The relevant functions that access these two maps are not expected to be called concurrently, but they are potential bugs without considering critical sections. I think concurrent access guard must be considered too. However, this change must be cautiously implemented, and enough test cases too. How about moving this consideration to another PR?

Reply:
@hyunsooda I agree with you.

Reply:
The topic in this conversation is handled in https://github.com/klaytn/klaytn/pull/1438